app/init/analytics.ts

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import DeviceInfo from 'react-native-device-info';
+import {Dimensions} from 'react-native';
+
+import LocalConfig from '@assets/config.json';
+import {Config} from '@mm-redux/types/config';
+import tracker from '@utils/time_tracker';
+
+type RudderClient = {
+    setup(key: string, options: any): Promise<void>;
+    track(event: string, properties: Record<string, any> | undefined, options?: Record<string, any>): void;
+    identify(userId: string, traits: Record<string, any>, options?: Record<string, any>): Promise<void>;
+    screen(name: string, properties: Record<string, any> | undefined, options?: Record<string, any>): void;
+    reset(): Promise<void>;
+}
+
+let diagnosticId: string | undefined;
+export let analytics: RudderClient | null = null;
+export let context: any;
+
+export async function init(config: Config) {
+    if (!analytics) {
+        analytics = require('@rudderstack/rudder-sdk-react-native').default;
+    }
+
+    if (analytics) {
+        const {height, width} = Dimensions.get('window');
```

Comment:
These opposite conditions feel weird to me. Is the `if (analytics)` check necessary?

Reply:
Ha, good catch, It seems that I’ve missed this and left it as it was, will remove the if statement as soon as I can, not a blocker though

Reply:
@chuttam so the reason for this is actually that TS fails with
> app/init/analytics.ts(32,15): error TS2531: Object is possibly 'null'.

by wrapping in the if statement the error goes away, so I'm going to leave it there